File: src/main/java/me/devsnox/battlekits/kits/entites/KitPlayer.java
package me.devsnox.battlekits.kits.entites;

import me.devsnox.battlekits.kits.kit.BattleKit;
import me.devsnox.battlekits.kits.kit.PlayerKit;

import java.util.HashMap;
import java.util.UUID;

public class KitPlayer {

    private UUID uuid;

    private HashMap<Integer, PlayerKit> kits;

    public KitPlayer(UUID uuid, HashMap<Integer, PlayerKit> kits) {
        this.uuid = uuid;
        this.kits = kits;
    }

    public void addKit(BattleKit battleKit) {
        this.kits.put(battleKit.getId(), new PlayerKit(battleKit.getId(), battleKit, 0));
    }

    public void removeKit(BattleKit battleKit) {
        this.kits.remove(battleKit.getId());
    }

    public UUID getUuid() {
        return uuid;
    }

    public HashMap<Integer, PlayerKit> getKits() {
        return kits;
    }
}

File: spigot/src/main/kotlin/me/devsnox/battlekits/kits/entites/KitPlayer.kt
package me.devsnox.battlekits.kits.entites

import me.devsnox.battlekits.functions.KitsP
import me.devsnox.battlekits.kits.kit.BattleKit
import me.devsnox.battlekits.kits.kit.PlayerKit
import java.util.*

class KitPlayer(val uuid: UUID, val kits: KitsP) {

    fun addKit(battleKit: BattleKit) {
        kits[battleKit.id] = PlayerKit(battleKit.id, battleKit, 0)
    }

    fun removeKit(battleKit: BattleKit) {
        kits -= battleKit.id
    }
}
